src/2025h2/reflection-and-comptime.md

```diff
@@ -0,0 +1,131 @@
+# reflection and comptime
+
+| Metadata         |                                                                                  |
+|:-----------------|----------------------------------------------------------------------------------|
+| Point of contact | @oli-obk                                                                         |
+| Teams            | <!-- TEAMS WITH ASKS -->                                                         |
+| Task owners      | <!-- TASK OWNERS -->                                                             |
+| Status           | Proposed                                                                         |
+| Other Tracking issue   | [rust-lang/rust#142577]                                  |
+| Zulip channel    | N/A (an existing stream can be re-used or new streams can be created on request) |
+
+## Summary
+
+Design, implement and experimentally land a reflection scheme based on `const fn` that can only be called at compile time.
+
+## Motivation & status quo
+
+Creating new general purpose crates (like serialization crates, log/tracing crates, game engine state inspection crates) that should work with almost all other data structures is nontrivial today. You either need to locally implement your traits for other crates, or the other crates need to depend on you and implement your traits. This often hinders rollout and will never reach everything.
+
+Reflection offers a way out of this dilemma, as you can write your logic for all types, by processing the type information at runtime (or even preprocess it at compile-time) without requiring trait bounds on your functions or trait impls anywhere.
+
+### The next 6 months
+
+* add an attribute for `const fn` that prevents them from being called from runtime code or `const fn` without the attribute
+* add basic datastructures to libcore that represent common information about types and the APIs to obtain that information
+
+### The "shiny future" we are working towards
+
+Create basic building blocks that allow `facet`, `bevy-reflect` and `reflect` to process types without requiring derives or trait bounds.
+
+## Design axioms
+
+* Prefer procedural const-eval code over associated const based designs
```

Comment:
I think there are interesting! I'm having a bit of trouble connecting them to the write-up above, perhaps because I don't quite get what alternatives you are weighing against.